source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.h

```diff
@@ -40,6 +41,14 @@ struct VariableBinding {
   std::string value;
 };
 
+struct MethodInfo {
+  const Protobuf::MethodDescriptor* descriptor_ = nullptr;
+  std::vector<const Protobuf::Field*> input_body_field_path;
+  bool input_type_is_http_body_ = false;
+  bool output_type_is_http_body_ = false;
+};
+typedef std::shared_ptr<MethodInfo> MethodInfoPtr;
```

Comment:
Unfortunately `google::grpc::transcoding::PathMatcher` returns `nullptr` on mismatch:
https://github.com/grpc-ecosystem/grpc-httpjson-transcoding/blob/82850792f42ab7ea7ee9b9c6bcbfc578bc3810c0/src/include/grpc_transcoding/path_matcher.h#L408

In addition to that (though minor) I'd like to avoid copying of `input_body_field_path` on every request.

Reply:
ah I see you changed the PathMatcher type param. OK, then just rename this to `MethodInfoSharedPtr`.

Reply:
oh, thanks